crypto/encrypt.go

```diff
@@ -26,67 +27,52 @@ func GenerateTwofishKey() (key TwofishKey, err error) {
 	return
 }
 
-// EncryptBytes encrypts a []byte using the key. The padded ciphertext, iv, and
-// amount of padding used are returned. `plaintext` is not overwritten.
-func (key TwofishKey) EncryptBytes(plaintext []byte) (ciphertext []byte, iv []byte, padding int, err error) {
-	// Determine the length needed for padding. The ciphertext must be padded
-	// to a multiple of twofish.BlockSize.
-	padding = twofish.BlockSize - (len(plaintext) % twofish.BlockSize)
-	if padding == twofish.BlockSize {
-		padding = 0
+// EncryptBytes encrypts a []byte using the key. EncryptBytes uses GCM and
+// prepends the nonce (12 bytes) to the ciphertext.
+func (key TwofishKey) EncryptBytes(plaintext []byte) (ct Ciphertext, err error) {
+	// Create the cipher, encryptor, and nonce.
+	twofishCipher, err := twofish.NewCipher(key[:])
+	if err != nil {
+		return nil, err
 	}
-
-	// Create the padded + unencrypted ciphertext.
-	ciphertext = make([]byte, len(plaintext)+padding)
-	copy(ciphertext, plaintext)
-
-	// Create the iv.
-	iv = make([]byte, twofish.BlockSize)
-	_, err = rand.Read(iv)
+	aead, err := cipher.NewGCM(twofishCipher)
 	if err != nil {
-		return
+		return nil, err
 	}
-
-	// Encrypt the ciphertext.
-	blockCipher, err := twofish.NewCipher(key[:])
+	nonce := make([]byte, aead.NonceSize())
+	_, err = rand.Read(nonce)
 	if err != nil {
-		return
+		return nil, err
 	}
-	encrypter := cipher.NewCBCEncrypter(blockCipher, iv)
-	encrypter.CryptBlocks(ciphertext, ciphertext)
 
-	return
+	// Encrypt the data. No authenticated data is provided, as EncryptBytes is
+	// meant for file encryption.
+	ct = append(nonce, aead.Seal(nil, nonce, plaintext, nil)...)
```

Comment:
I'm not 100% sure, but I think it's okay to write `aead.Seal(nonce, nonce, plaintext, nil)`. You should test it to make sure.